src/components/FeeControlGroup/index.tsx

```diff
@@ -0,0 +1,118 @@
+import { useEffect, useState } from 'react'
+import { useSearchParams } from 'react-router-dom'
+import { Flex, Text } from 'rebass'
+import styled, { css } from 'styled-components'
+
+import useGetFeeConfig from 'components/SwapForm/hooks/useGetFeeConfig'
+import { DEFAULT_TIPS } from 'constants/index'
+import useTheme from 'hooks/useTheme'
+
+import CustomFeeInput from './CustomFeeInput'
+
+const feeOptionCSS = css`
+  height: 100%;
+  padding: 0;
+  border-radius: 20px;
+  border: 1px solid transparent;
+
+  background-color: ${({ theme }) => theme.tabBackground};
+  color: ${({ theme }) => theme.subText};
+  text-align: center;
+
+  font-size: 12px;
+  font-weight: 400;
+  line-height: 16px;
+
+  outline: none;
+  cursor: pointer;
+
+  :hover {
+    border-color: ${({ theme }) => theme.bg4};
+  }
+  :focus {
+    border-color: ${({ theme }) => theme.bg4};
+  }
+
+  &[data-active='true'] {
+    background-color: ${({ theme }) => theme.tabActive};
+    color: ${({ theme }) => theme.text};
+    border-color: ${({ theme }) => theme.primary};
+
+    font-weight: 500;
+  }
+`
+
+const DefaultFeeOption = styled.button`
+  ${feeOptionCSS};
+  flex: 0 0 18%;
+
+  @media only screen and (max-width: 375px) {
+    font-size: 10px;
+    flex: 0 0 15%;
+  }
+`
+
+const FeeControlGroup = () => {
+  const theme = useTheme()
+  const { feeAmount, enableTip } = useGetFeeConfig() ?? {}
+  const [searchParams, setSearchParams] = useSearchParams()
+
+  const [feeValue, setFee] = useState(Math.round(Number.parseFloat(feeAmount ?? '0')) || 0)
```

Comment:
why do we need this state? 
can use searchParams.get('feeAmount') directly.

Reply:
Updated